compiler/src/dotty/tools/dotc/classpath/ClassPath.scala

```diff
@@ -52,7 +52,8 @@ sealed trait BinaryFileEntry extends ClassRepresentation {
 object BinaryFileEntry {
   def apply(file: AbstractFile): BinaryFileEntry =
     if file.isTasty then
-      TastyFileEntry(file)
+      if file.resolveSiblingWithExtension("class") != null then TastyFileEntry(file)
```

Comment:
Should we rename `TastyFileEntry` into `TastyAndClassFileEntry` then?

Reply:
`TastyWithClassFileEntry` might be more appropriate.

Reply:
Updated